RHEcosystemAppEng/dbaas-operator
# [Dbaas-1060] DBaaS instance & DBaaS Provider metrics

## Description
Rework DBaaSInstance/DBaaSProvider metrics based on the design proposal at this [link](https://docs.google.com/presentation/d/1R9jdTdBwDp70Xlr9r_83nsbG7JUqA8I0DsnKE_N3hsk/edit?usp=sharing)

Screenshots
<img width="1266" alt="image" src="https://user-images.githubusercontent.com/87712456/203828323-d7266357-1f52-4c9d-8060-0c74871aa24c.png">

<img width="1207" alt="image" src="https://user-images.githubusercontent.com/87712456/203828526-cfbd9cce-e7f0-41b1-a631-d760cdb7ca74.png">

<img width="1207" alt="image" src="https://user-images.githubusercontent.com/87712456/203828585-179fe3cd-56bf-4958-b736-fdedc93b5563.png">

## Verification Steps(OpenShift Local)

- Install the DBaaS Operator on a fresh OpenShift cluster.
- After successful installation expose the Prometheus service

```
oc project openshift-dbaas-operator
oc expose svc/prometheus-operated
oc get route 
```

- Navigate to Prometheus Service Route in the browser, select a metric such as "dbaas_requests_duration_seconds_sum" and see the resource="dbaas_provider" appear.
- Register MongoDB Provider and create a new MongoDB Instance by following DBaaS Operator User Guide. In the prometheus console, see the resource="dbaas_instance" appear.
- Note: The DBaaSProvider does not have delete handler in the controller, so the delete event is not possible.

## Type of change


## Checklist (Definition of Done)
